extensions/default/src/DicomWebDataSource/index.js

```diff
@@ -432,13 +442,21 @@ function createDicomWebApi(dicomWebConfig, userAuthenticationService) {
 
       DicomMetadataStore.addSeriesMetadata(seriesSummaryMetadata, madeInClient);
 
-      const seriesDeliveredPromises = seriesPromises.map(promise =>
-        promise.then(instances => {
+      const seriesDeliveredPromises = seriesPromises.map(promise => {
+        if (!returnPromises) {
+          promise?.start();
+        }
+        return promise.then(instances => {
           storeInstances(instances);
-        })
-      );
-      await Promise.all(seriesDeliveredPromises);
-      setSuccessFlag();
+        });
+      });
+      if (returnPromises) {
```

Comment:
@rodrigobasilio2022 comment for you

Reply:
I don't think this is ugly because in mode.tsx this function is not called directly: there is another function that checks in the server configuration if lazy study loading is ok. If it's not, it will retrieve data synchronously, not considering this function. So either way I will need to check the returning value of the async function to see if the series promises are defined to use them. The way implemented in this PR changed the code the minimum, to try to avoid inserting some regression bugs.

Reply:
@sedghi - I really dislike the returning of different types/paths here because it means you need to know how the function is behaving fundamentally to know whether you can wait on various elements.  What I was suggestion was to always return the same type of value, but to have attributes of the returned promise available to get sub-parts of the promise.    That is, make series.metadata return a Promise<Map<string,Promise<SeriesMetadata>>> 
Then, only for the Mode version of it, if the resolution of the promise is a Map, then it contains the additional promises, which are then sub-selected to wait for the required set - all the time, based just on the hanging protocols defined/being applied, and if it isn't a Map, then don't wait for anything inside the nested promise.

Reply:
I agree this is an important point and we should wait for @sedghi comments